server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -120,6 +123,12 @@
                 + org.apache.lucene.util.Version.LATEST + "] is still set to [" + CURRENT.luceneVersion + "]";
 
         idToVersion = builder.build();
+        ServiceLoader<CompatibleApiVersionProvider> load = ServiceLoader.load(CompatibleApiVersionProvider.class);
+        Optional<CompatibleApiVersionProvider> first = load.findFirst();
+        CompatibleApiVersionProvider compatibleApiVersionProvider = first.orElseGet(() -> () -> Version.CURRENT);
+        // TODO should we fetch that dynamically when calling minimumRestCompatibilityVersion() ?
+        minimumRestCompatibilityVersion = compatibleApiVersionProvider.minimumRestCompatibilityVersion();
+        System.out.println(minimumRestCompatibilityVersion);
```

Comment:
this is nice to have ... but I don't think it is needed since it will always be current major-1. If i understand correctly this also requires an additional interface ?

Can we remove this ?